src/utils/context.ts

```diff
@@ -4,17 +4,27 @@ import { CowError, logPrefix } from './common'
 import { SupportedChainId as ChainId } from '../constants/chains'
 import { DEFAULT_APP_DATA_HASH, DEFAULT_IPFS_GATEWAY_URI } from '../constants'
 
+export interface Ipfs {
+  uri?: string
+  apiKey?: string
+  apiSecret?: string
```

Comment:
Would it work to pin the file using any other IPFS provider?
To be honest I don't know any besides Pinata, but there should be more options, right?
In any way, we are not generic and are bound to Pinata in the code. I think it would be better to make that fact more clear in the param names and the README so consumers know we are expecting Pinata api info